hw_diag/utilities/miner.py

```diff
@@ -33,3 +34,8 @@ def fetch_miner_data(diagnostics):
     diagnostics['MN'] = peerbook.get('nat')
     diagnostics['MR'] = diagnostics['MN'] == 'symmetric'
     return diagnostics
+
+
+def create_add_gateway_txn(destination_wallet):
+    add_gateway_txn = client.create_add_gateway_txn(destination_wallet, NEBRA_WALLET_ADDRESS)
```

Comment:
Not a blocker but the code at line 7 `client = MinerClient()` and similar pattern in other files should usually be defined inside the class/function that makes use of it. In this case, this line should have been inside both `fetch_miner_data` and `create_add_gateway_txn`.

The reasoning: Module level instances get created on module import and since they are module-level-global there are chances of more things going wrong with them. Like if one of the calling functions causes them to be in an invalid state then all other functions using them would also suffer.

One of the exceptions to this rule is singletons and entry points for web frameworks. Like the `app` object for Flask.